homeassistant/components/light/mipow.py

```diff
@@ -0,0 +1,194 @@
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_EFFECT, EFFECT_RAINBOW)
+from homeassistant.components.light import (
+    EFFECT_CANDLE, ATTR_FLASH, FLASH_LONG, FLASH_SHORT, Light)
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pygatt[GATTTOOL]==3.0.0', 'pexpect==4.0.1']
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Add device specified by serial number."""
+    serial = config['serial']
+    name = config.get('name')
+    bulb = Mipow(serial, name)
+
+    add_devices_callback([bulb])
+
+
+class Mipow(Light):
+    """Main class."""
+
+    def __init__(self, serial, name=None):
+        """Initialize the light."""
+        if name is not None:
+            self._name = name
+        else:
+            self._name = serial
+
+        self._serial = serial
+        self._adapter = None
+        self._connection = None
+        try:
+            self.update()
+        except:
+            self._rgb_color = [0, 0, 0]
+            self._rgb_bright = [0]
+
+    def update(self):
+        """Read back the device state."""
+        self._rgb_color = self.rgb_color
+        self._rgb_bright = self.rgb_bright
+
+    def _start_adapter(self):
+        """Start the adapter."""
+        import pygatt
+        if self._adapter is not None:
+            self._adapter.stop()
+
+        adapter = pygatt.backends.GATTToolBackend()
+        adapter.start()
+
+        self._adapter = adapter
+
+        return adapter
+
+    def _stop_adapter(self):
+        """Stop the adapter."""
+        self._adapter.stop()
+
+    def connect(self):
+        """Connect to lamp."""
+        import pygatt
+        if self._connection is not None:
+            if self._connection._connected:
+                return self._connection
+        else:
+            if self._adapter is not None:
+                self._adapter.stop()
+
+            adapter = pygatt.backends.GATTToolBackend()
+            adapter.start()
+
+            self._adapter = adapter
+            self._connection = self._adapter.connect(self._serial)
+            return self._connection
+
+    @property
+    def should_poll(self):
+        """Polling needed."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._name
+
+    @property
+    def rgb_color(self):
+        """Read rgb color."""
+        self._connection = self.connect()
+
+        device_status = (
+            self._connection.char_read("0000fffc-0000-1000-8000-00805f9b34fb"))
+
+        device_colors = [x for x in device_status]
+
+        device_colors.pop(0)
+
+        self._rgb_color = device_colors
+
+        return self._rgb_color
+
+    @property
+    def rgb_bright(self):
+        """Read brightness."""
+        self._connection = self.connect()
+
+        device_status = (
+            self._connection.char_read("0000fffc-0000-1000-8000-00805f9b34fb"))
```

Comment:
What is this string? It would be better if it were an appropriately named 'constant' at the top of this file.